nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/document/ExtractDocumentText.java

```diff
@@ -0,0 +1,142 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.document;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.tika.Tika;
+import org.apache.tika.exception.TikaException;
+
+import java.io.BufferedInputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Tags({"extract, text, pdf, word, excel, powerpoint, office"})
+@CapabilityDescription("Run Apache Tika text extraction to extra the text from supported binary file formats such as PDF " +
+        "and Microsoft Office files.")
+public class ExtractDocumentText extends AbstractProcessor {
+    private static final String TEXT_PLAIN = "text/plain";
+
+    public static final String FIELD_MAX_TEXT_LENGTH = "MAX_TEXT_LENGTH";
+    public static final String FIELD_SUCCESS = "success";
+    public static final String FIELD_FAILURE = "failure";
+
+    public static final PropertyDescriptor MAX_TEXT_LENGTH = new PropertyDescriptor.Builder()
+            .name(FIELD_MAX_TEXT_LENGTH)
+            .displayName("Max Output Text Length")
+            .description("The maximum length of text to retrieve. This is used to limit memory usage for " +
+                    "dealing with large files. Specify -1 for unlimited length.")
+            .required(false).defaultValue("-1").addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES).build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name(FIELD_SUCCESS)
+            .description("Successfully extract content.").build();
```

Comment:
Recommend removing period character:
```suggestion
            .description("Content extraction success").build();
```